site/cypress/e2e/Cypress-UI/navigation_page.spec.js

```diff
@@ -9,6 +7,91 @@ function validate_navigation_page_sections(sections) {
     });
 }
 
+const white = 'rgb(255, 255, 255)';
+const blue = 'rgb(0, 99, 152)';
+const red = 'rgb(220, 53, 69)';
+const green = 'rgb(95, 183, 96)';
+const buttonOrder = ['team-btn', 'submit-btn', 'grade-btn', 'quick-link-btn'];
+
+function checkButtons(courseID, buttonText, buttonColor) {
+    // assert that the course does not exist
+    if (buttonText === null && buttonColor === null) {
+        cy.get(`[data-testid="${courseID}"]`).should('not.exist');
+    }
+    else {
+        for (let i = 0; i < buttonOrder.length; i++) {
+            if (buttonText[i]) {
+                cy.get(`[data-testid="${courseID}"]`).find(`[data-testid="${buttonOrder[i]}"]`).contains(buttonText[i]);
+                cy.get(`[data-testid="${courseID}"]`).find(`[data-testid="${buttonOrder[i]}"]`).should('have.css', 'background-color', buttonColor[i]);
+            }
+            else {
+                // assert that the button does not exist
+                cy.get(`[data-testid="${courseID}"]`).find(`[data-testid="${buttonOrder[i]}"]`).should('not.exist');
+            }
+        }
+    }
+}
+
+describe('tests navigation buttons for each level of access', () => {
+    beforeEach(() => {
+        cy.visit('/');
```

Comment:
In this case you don't need to use `cy.visit()` before the tests, as `cy.login` will then redirect, and so will cy.visit(sample) afterwards.

Also, you can just use `cy.visit();` to visit the homepage.